packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx

```diff
@@ -0,0 +1,278 @@
+  const [selectState, setSelectState] = useState<SelectState>({});
+  const [collections, setCollections] = useState<Tree[]>([]);
+
+  const selectDatabase = useMemo(() => {
+    return (database: string): void => {
+      setSelectState({
+        database: database ?? undefined,
+      });
+      const items = tree.find((x) => x.value === database)?.items;
+      setCollections(items || []);
+    };
+  }, [tree]);
+
+  const selectCollection = useMemo(() => {
+    return (collection: string): void => {
+      setSelectState({
+        ...selectState,
+        collection: collection ?? undefined,
+      });
+    };
+  }, [selectState]);
+
+  const treeControls = useMemo(() => {
+    return (
+      <div className={selectContainer}>
+        <RenderFilterSelect
+          items={tree}
+          placeHolder={'All databases'}
+          onSelect={selectDatabase}
+          defaultValue={selectState.database}
+        />
+        <RenderFilterSelect
+          items={collections}
+          placeHolder={'All collections'}
+          onSelect={selectCollection}
+          defaultValue={selectState.collection}
+        />
+      </div>
+    );
+  }, [selectDatabase, tree, collections, selectCollection, selectState]);
+
+  return [treeControls, selectState];
+}
+
+function convertItemsToTree(items: Item[]): Tree[] {
+  const dbCollectionMap: Record<string, string[]> = {};
+  const tree = [];
+  items.forEach((item) => {
+    if (!dbCollectionMap[item.database]) {
+      dbCollectionMap[item.database] = [];
+    }
+    dbCollectionMap[item.database].push(item.collection);
+  });
+  for (const database in dbCollectionMap) {
+    const collections = dbCollectionMap[database]
+      .filter((collection, index, arr) => arr.indexOf(collection) === index)
+      .map((collection) => ({
+        value: collection,
+      }));
+
+    tree.push({
+      value: database,
+      items: collections,
+    });
+  }
+  return tree;
+}
+
+function getSearchableText(item: Item) {
+  let searchable: unknown[] = [item.name, item.database, item.collection];
+
+  if (item.type === 'aggregation') {
+    const stages = item.aggregation.pipeline
+      .filter((x) => x.stageOperator && x.stage)
+      .map((stage) => {
+        return {
+          [stage.stageOperator]: JSON.parse(stage.stage),
+        };
+      });
+    searchable = searchable.concat([
+      item.aggregation.namespace,
+      item.aggregation.env ?? '',
+      item.aggregation.collationString ?? '',
+      item.aggregation.isTimeSeries ? ['timeseries', 'time series'] : '',
+      ['aggregate', 'aggregation'],
+      stages,
+    ]);
+  } else {
+    searchable = searchable.concat([
+      item.query._ns,
+      ['find', 'query'],
+      {
+        filter: item.query.filter,
+        project: item.query.project,
+        sort: item.query.sort,
+        skip: item.query.skip,
+        limit: item.query.limit,
+        collation: item.query.collation,
+      },
```

Comment:
IIUC this means that typing something like `limit` will match every query in the list, is this expected?

Reply:
after discussing this @mcasimir, for queries we are just using filter to search on.